ログイン機能の実装

halki117 opened the pull request:
- ユーザーの新規登録機能、ログイン機能、及びログアウト機能を実装いたしました。
- sourceryというgemを使用しての実装となります

halki117 committed:
first_commit

halki117 committed:
test

halki117 committed:
socery01

halki117 committed:
soecery_02

halki117 committed:
sorcery_03

halki117 committed:
insta_04

halki117 committed:
insta_05

halki117 committed:
fixed

halki117 committed:
fixed

halki117 committed:
font-awesome

halki117 committed:
view_fix

halki117 committed:
flash実装

flashメッセージの実装を行いました

halki117 committed:
view_imp

主にヘッダーのビューの実装

halki117 committed:
view-imp2

halki117 committed:
fix

halki117 committed:
controller_fix

DaichiSaito commented on .rubocop.yml:
```diff
@@ -0,0 +1,81 @@
+
+# メソッドの行数が多すぎないかをチェック
+Metrics/MethodLength:
+ Max: 30
+
+# ネストが深すぎないかをチェック（if文のネストもチェック）
+Metrics/BlockNesting:
+ Max: 5
+
+# クラスの行数をチェック（無効）
+Metrics/ClassLength:
+ Enabled: false
+
+# 空メソッドの場合に、1行のスタイルにしない　NG例：def style1; end
+Style/EmptyMethod:
+ EnforcedStyle: expanded
+
+# クラス内にクラスが定義されていないかチェック（無効）
+Style/ClassAndModuleChildren:
+ Enabled: false
+
+# 日本語でのコメントを許可
+Style/AsciiComments:
+ Enabled: false
+
+# クラスやモジュール定義前に、それらの説明書きがあるかをチェック（無効）
+Style/Documentation:
+ Enabled: false
+
+# ％i（）構文を使用していないシンボルで構成される配列リテラルをチェック（無効）
+Style/SymbolArray:
+ Enabled: false
+
+# 文字列に値が代入されて変わっていないかチェック（無効）
+Style/FrozenStringLiteralComment:
+ Enabled: false
+
+# メソッドパラメータ名の最小文字数を設定
+Naming/MethodParameterName:
+ MinNameLength: 1
```
すごい、めっちゃ詳しく書いてる！

ここは現場によって設定が違うのであくまでも参考程度に！覚えておかないといけないのはrubocopがないポートフォリオは初学者感満載ということ。

halki117 replied:
これ某記事からちょっと拝借したものなんです^^;
ポートフォリオにもrubocopを忘れない様にします！

DaichiSaito commented on app/assets/javascripts/application.js:
```diff
@@ -0,0 +1,19 @@
+//= require activestorage
+//= require_tree .
+
+//  require turbolinks を外している!!!
```
rails newのときのオプションでそもそもturbolinks外せますよ〜

halki117 replied:
--skip-turbolinks で外せるみたいですね、了解です！

DaichiSaito commented on app/controllers/sessions_controller.rb:
```diff
@@ -0,0 +1,23 @@
+class SessionsController < ApplicationController
+  skip_before_action :require_login, except: [:destroy]
+
+  def new
+    @user = User.new
+  end
```
解答例はこうなってましたか？

halki117 replied:
よく思えば view では @user を使って無いですね、
@user = User.new は削除しました。

DaichiSaito commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,22 @@
+  end
+
+  def create
+    if @user.save
+    user = User.new(user_params)
+    if user.save
+      redirect_to root_path
+    else
+      render :new
+    end 
```
なんかSyntaxErrorが発生しそうな気が。

halki117 replied:
if @user.save の部分が不要だったので削除しました。

DaichiSaito commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,22 @@
+
+  private
+  def user_params
+    params.require(:user).permit(:name, :email ,:password ,:password_confirmation,)
```
最後カンマ不要

halki117 replied:
カンマ削除完了です。

DaichiSaito commented on app/helpers/posts_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module PostsHelper
+end
```
helperを生成しないような設定があるのでそれを設定すべきではあったかな。
また、使わないhelperは削除しちゃった方がいいと思いますよ。見辛いので。

halki117 replied:
かしこまりました。
config/application.rb 内に設定を記入して helper ついでに coffee も生成しないようにしました。

DaichiSaito commented on app/helpers/application_helper.rb:
```diff
@@ -0,0 +1,9 @@
+module ApplicationHelper
+
+  def avatar_url(user)
+    return user.profile_photo unless user.profile_photo.nil?
+    "https://image.shutterstock.com/image-vector/sample-stamp-square-grunge-sign-260nw-1474408826.jpg"
+  end
```
これはなんのために追加しましたか？

halki117 replied:
もしユーザーのプロフィール画像のカラムが空ならデフォルトの画像を表示する意味で追加してました。

halki117 committed:
bootstrap-material-design

bootstrap-material-designがなぜか効かなくなってたので修正
下記、記事を参照
https://qiita.com/kenkentarou/items/e2ee6062fbff5d69fffd

halki117 committed:
fix

halki117 committed:
fix

halki117 committed:
fix
